### ISSUE TITLE
How to set learning rate group for variables in existing VarStore

### ISSUE THREAD

**Issue by tobygodwin** [LaurentMazare/tch-rs#584]
Hello,

Firstly thank you very much for your great work maintaining this crate! 

I'm trying to load  and train a model created with JIT, but assign different learning rates to different variables in the model.

Following this [example](https://github.com/LaurentMazare/tch-rs/blob/main/examples/jit-train/main.rs), I define a `VarStore` and then load a `TrainableCModule`, which adds the tensors from the JIT module to the `VarStore` path.

```let vs = VarStore::new(device);
let mut trainable = TrainableCModule::load("model.pt", vs.root())?;
```
At this point, I'd like to assign different learning rates to different variables in the VarStore

I have tried to create a new VarStore and copy over variables while assigning groups (as below), but it seems that these weights are not updated during the optimization step (which makes sense given we add all the tensors from the JIT module to total_vs)
```
let total_vs = VarStore::new(device);
let vs = VarStore::new(device);
let mut trainable = TrainableCModule::load("model.pt", total_vs.root())?;
...
for (var_name, var_tensor) in total_vs.variables() {
       if var_name.starts_with("group_0_prefix") {
             vs.root().set_group(0).f_var_copy(&var_name, &var_tensor);
       } 
       else {
             vs.root().set_group(1).f_var_copy(&var_name, &var_tensor);
       }
 }
...
opt.set_lr_group(1, lr1 as f64);
opt.set_lr_group(0, lr2 as f64);
```

My question is how can I train different variables in `total_vs` with different learning rates but using the same optimizer?

Thanks very much for your time, 
Toby

**Comment by LaurentMazare** [comment]
I think you should be able to do this by using the `CModule` directly and not through `TrainableCModule`. E.g. rather than creating the module via the [load function](https://github.com/LaurentMazare/tch-rs/blob/9782e497ba71a1da710ad4567d5f5d4edc79f609/src/wrappers/jit.rs#L636-L642). You can instead try something along the lines of:
```rust
        let vs = VarStore::new(device);
        let cmodule = CModule::load_on_device(module_path, path.device())?;
        for (name, tensor) in cmodule.named_parameters()? {
            let requires_grad = tensor.requires_grad();
            let path = ...; // Use different groups here based on name?
            let _t = path.add(&name.replace('.', "_"), tensor, requires_grad);
        }
        let mut opt = Adam::default().build(&vs, 1e-4)?;
```
I haven't tested it though, so let me know if it doesn't work and I'll actually try it.
